src/NSwag.SwaggerGeneration.WebApi/Processors/SwaggerResponseBuilder.cs

```diff
@@ -0,0 +1,218 @@
+                if (attributeType.GetRuntimeProperty("Description") != null)
+                {
+                    if (!string.IsNullOrEmpty(responseTypeAttribute.Description))
+                        description = responseTypeAttribute.Description;
+                }
+
+                var isNullable = true;
+                if (attributeType.GetRuntimeProperty("IsNullable") != null)
+                    isNullable = responseTypeAttribute.IsNullable;
+
+                OperationResponseModels.Add(new OperationResponseModel(httpStatusCode, returnType, isNullable, description));
+            }
+        }
+
+        public void PopulateModelsFromProducesAttributes(IEnumerable<Attribute> producesResponseTypeAttributes)
+        {
+            foreach (dynamic producesResponseTypeAttribute in producesResponseTypeAttributes)
+            {
+                var returnType = producesResponseTypeAttribute.Type;
+                var httpStatusCode = producesResponseTypeAttribute.StatusCode.ToString(CultureInfo.InvariantCulture);
+                var description = HttpUtilities.IsSuccessStatusCode(httpStatusCode) ? _successXmlDescription : string.Empty;
+                OperationResponseModels.Add(new OperationResponseModel(httpStatusCode, returnType, true, description));
+            }
+        }
+
+
+        public async Task BuildSwaggerResponseAsync(
+            ParameterInfo returnParameter)
+        {
+            
+            foreach (var statusCodeGroup in OperationResponseModels.GroupBy(r => r.HttpStatusCode))
+            {
+                var httpStatusCode = statusCodeGroup.Key;
+                var returnType = statusCodeGroup.Select(r => r.ResponseType).FindCommonBaseType();
+                var description = string.Join("\nor\n", statusCodeGroup.Select(r => r.Description));
+
+                var typeDescription = _settings.ReflectionService.GetDescription(
+                    returnType, returnParameter?.GetCustomAttributes(), _settings);
+
+                var response = new SwaggerResponse
+                {
+                    Description = description ?? string.Empty
+                };
+
+                if (IsVoidResponse(returnType) == false)
+                {
+                    response.IsNullableRaw = OperationResponseModels.Any(r => r.IsNullable) && typeDescription.IsNullable;
+                    response.ExpectedSchemas = await GenerateExpectedSchemasAsync(statusCodeGroup);
+                    response.Schema = await _context.SchemaGenerator
+                        .GenerateWithReferenceAndNullability<JsonSchema4>(
+                            returnType, null, typeDescription.IsNullable, _context.SchemaResolver)
+                        .ConfigureAwait(false);
+                }
+
+                _context.OperationDescription.Operation.Responses[httpStatusCode] = response;
+            }
+
+            bool loadDefaultSuccessResponseFromReturnType;
+            if (OperationResponseModels.Count > 0)
```

Comment:
Slightly tweaked the previous code. This is what the condition looked like in `OperationParameterProcessor`:

```C#
if (responseTypeAttributes.Any() || producesResponseTypeAttributes.Any())
{
    ...
    var operationResponses = context.OperationDescription.Operation.Responses;
    var hasSuccessResponse = operationResponses.ContainsKey("200") || operationResponses.ContainsKey("204");
 
    var loadDefaultSuccessResponseFromReturnType = !hasSuccessResponse &&
        context.MethodInfo.GetCustomAttributes()
            .Any(a => a.GetType().IsAssignableTo("SwaggerDefaultResponseAttribute", TypeNameStyle.Name)) ||
        context.MethodInfo.DeclaringType.GetTypeInfo().GetCustomAttributes()
            .Any(a => a.GetType().IsAssignableTo("SwaggerDefaultResponseAttribute", TypeNameStyle.Name));
 
    if (loadDefaultSuccessResponseFromReturnType)
        await LoadDefaultSuccessResponseAsync(context.OperationDescription.Operation, context.MethodInfo, 
            successXmlDescription, context.SchemaGenerator, context.SchemaResolver).ConfigureAwait(false);
}
else
    await LoadDefaultSuccessResponseAsync(context.OperationDescription.Operation, context.MethodInfo, 
        successXmlDescription, context.SchemaGenerator, context.SchemaResolver).ConfigureAwait(false);
```